### PR TITLE
8262235: Remove unnecessary logic in hugetlbfs_sanity_check()

### PR DESCRIPTION
**Pull request by kstefanj**
Please review this change that removes some additional verification not really needed when doing the sanity check for hugetlbfs-pages.

**Summary**
The call to `mmap` in `hugetlbfs_sanity_check()` will only succeed if it can satisfy the mapping using large pages. If this was not the case, we would have to double check all large page mappings not just this sanity check. So the reading of `/proc/self/maps` is safe to remove.

**Testing**
Sanity run through mach 5 and local testing with and without large pages configured.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262235](https://bugs.openjdk.java.net/browse/JDK-8262235): Remove unnecessary logic in hugetlbfs_sanity_check()

### Reviewers
 * [Ivan Walulya](https://openjdk.java.net/census#iwalulya) (@walulyai - Committer) ⚠️ Review applies to 2ef2316c91e8ff4b59f47b9e827d18de0e1e0fac
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3048/head:pull/3048`
`$ git checkout pull/3048`

To update a local copy of the PR:
`$ git checkout pull/3048`
`$ git pull https://git.openjdk.java.net/jdk pull/3048/head`

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -3482,39 +3482,21 @@ int os::Linux::hugetlbfs_page_size_flag(size_t page_size) {
...
-    }
+    // Mapping succeede, sanity check passed.
     munmap(p, page_size);
+    return true;
   }
 
-  if (warn && !result) {
-    warning("HugeTLBFS is not supported by the operating system.");
+  if (warn) {
+    warning("HugeTLBFS is not supported or configured by the operating system.");
```
Comment (walulyai) [inline src/hotspot/os/linux/os_linux.cpp:3496]: minor: "HugeTLBFS is not configured or not supported by the operating system."

**Review by walulyai — APPROVED** [review]
lgtm!

File: src/hotspot/os/linux/os_linux.cpp (hunk 1 above)
Comment (kstefanj) [reply src/hotspot/os/linux/os_linux.cpp:3496]: Thanks, I'll update it per your suggestion.
```suggestion
    warning("HugeTLBFS is not configured or not supported by the operating system.");
```

File: src/hotspot/os/linux/os_linux.cpp [hunk 2]
```diff
@@ -3482,39 +3482,21 @@ int os::Linux::hugetlbfs_page_size_flag(size_t page_size) {
...
-            if (strstr (chars, "hugepage")) {
-              result = true;
-              break;
-            }
-          }
-        }
-      }
-      fclose(fp);
-    }
+    // Mapping succeede, sanity check passed.
```
Comment (kstefanj) [inline src/hotspot/os/linux/os_linux.cpp:3490]:
```suggestion
    // Mapping succeeded, sanity check passed.
```

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Comment by kstefanj** [comment]
Thanks @walulyai and @tschatzl for the reviews.
/integrate

**Pull request closed without merging**